### ISSUE TITLE
removeTrack fails with 'Sender was not created by this connection.'

### ISSUE THREAD

**Issue by DanEdwards61** [webrtcHacks/adapter#624]
**Browsers and versions affected**
Chrome: Version 59.0.3071.115 (Official Build) (64-bit)
adapter.js: 4.2.0

**Description**
Calling removeTrack causes the exception 'Sender was not created by this connection.'.
The code that's failing is: 
      var isLocal = sender._pc === pc;
      if (!isLocal) {
        throw new DOMException('Sender was not created by this connection.',
            'InvalidAccessError');
      }

sender._pc is undefined. The only place _pc appears to be set is in the routine shimGetSendersWithDtmf.

**Steps to reproduce**
Call removeTrack

**Expected results**
Track should be removed without throwing an exception

**Actual results**
An exception is thrown.

**Comment by fippo** [comment]
how are you getting the sender? Ideally, can you show a fiddle that reproduces the issue?

**Comment by jan-ivar** [comment]
@DanEdwards61 [removeTrack](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/removeTrack#Syntax) expects a *sender*, not a track.

Though adapter should give a better error. Here's a [fiddle](https://jsfiddle.net/jib1/w6jqtchh/) to reproduce it.

In Chrome (w/adapter): InvalidAccessError: Sender was not created by this connection.
In Firefox: TypeError: Argument 1 of RTCPeerConnection.removeTrack does not implement interface RTCRtpSender.

Looks like we need an
```js
if (!(sender instanceof RTCRtpSender)) throw new TypeError(msgLikeInFF);
```

**Comment by jan-ivar** [comment]
I mean the same is true for most other polyfills, but given `replaceTrack`'s confusing name, I think a PR is warranted here.

**Comment by fippo** [comment]
will be fixed in #625 with a better error and test. Thanks!
